src/datetime/mod.rs

```diff
@@ -329,6 +330,19 @@ impl<Tz: TimeZone> DateTime<Tz> {
         Some(tz.from_utc_datetime(&datetime))
     }
 
+    /// Adds given `Months` to the current date and time.
+    ///
+    /// Returns `None` when it will result in overflow, or if the existing
```

Comment:
I feel like the "existing" word here doesn't add much value and distracts from the explanation. And maybe use "not valid for" instead of "not valid on"? Not a native speaker, so you may have a better idea than me.

Reply:
Removed the "existing" - agreed that didn't add much.

With the on/for, I'd lean toward "on" for a specific date, and "for" when talking about a longer period like a week. Did a quick search found only this: https://english.stackexchange.com/questions/182660/on-vs-at-with-date-and-time